src/test/kotlin/sirius/kernel/commons/StringsTest.kt

```diff
@@ -332,4 +332,90 @@ class StringsTest {
+        assertEquals("ABCDEFG", Strings.truncateMiddle("ABCDEFG", 10, 0))
+        assertEquals("ABCDEFG", Strings.truncateMiddle("ABCDEFG", 10, -10))
+        // Szenario 4: input present and length is smaller than input.length() and charactersToPreserveAtTheEnd is 0 or negative
+        assertEquals("ABCDEFG-AB…[truncated]", Strings.truncateMiddle("ABCDEFG-ABCDEFG", 10, 0))
+        assertEquals("ABCDEFG-AB…[truncated]", Strings.truncateMiddle("ABCDEFG-ABCDEFG", 10, -10))
+        assertEquals("ABCDEFG-AB…[truncated]", Strings.truncateMiddle("ABCDEFG-ABCDEFG-ABCDEFG-ABCDEFG-ABCDEFG-ABCDEFG", 10, -10))
+        // Szenario 5: input present and length is equal to input.length() and charactersToPreserveAtTheEnd is 0 or negative
+        assertEquals("ABCDEFG", Strings.truncateMiddle("ABCDEFG", 7, 0))
+        assertEquals("ABCDEFG", Strings.truncateMiddle("ABCDEFG", 7, -10))
+        // Szenario 6: input present and length is 0 and charactersToPreserveAtTheEnd is 0 or negative
+        assertEquals("…[truncated]", Strings.truncateMiddle("ABCDEFG", 0, 0))
+        assertEquals("…[truncated]", Strings.truncateMiddle("ABCDEFG", 0, -10))
+        // Szenario 7: input present and length is negative and charactersToPreserveAtTheEnd is 0 or negative
+        assertEquals("…[truncated]", Strings.truncateMiddle("ABCDEFG", -10, 0))
+        assertEquals("…[truncated]", Strings.truncateMiddle("ABCDEFG", -10, -10))
+        // Szenario 8: input present and length is greater than input.length() and charactersToPreserveAtTheEnd is greater than length
+        assertEquals("ABCDEFG", Strings.truncateMiddle("ABCDEFG", 10, 10000))
+        assertEquals("ABCDEFG", Strings.truncateMiddle("ABCDEFG", 10, 11))
+        // Szenario 9: input present and length is greater than input.length() and charactersToPreserveAtTheEnd is greater than input.length()
+        assertEquals("ABCDEFG", Strings.truncateMiddle("ABCDEFG", 10, 8))
+        // Szenario 10: input present and length is greater than input.length() and charactersToPreserveAtTheEnd is equal to input.length()
+        assertEquals("ABCDEFG", Strings.truncateMiddle("ABCDEFG", 10, 7))
+        // Szenario 11: input present and length is smaller than input.length() and charactersToPreserveAtTheEnd is greater than length and length + charactersToPreserveAtTheEnd is smaller than input.length()
+        // Note: the ellipsis after the truncated signal is only added if there are characters to preserve at the end
+        // Note: the length of the truncated signal is only taken into account when limiting the input to length if length is greater than 13, otherwise the input will be truncated to the provided length and the truncated signal is added afterward
+        assertEquals("ABCDEFG-…[truncated]…This is the important part", Strings.truncateMiddle("ABCDEFG-ABCDEFG-This is the important part", 8, 26))
```

Comment:
same as above

Reply:
yeah we can discuss about that...
What should be the alternative behavior?
You just don't get the "truncated signal" although you set it to `true`?